frontend/src/pages/Admin/Subcontractors/index.tsx

```diff
@@ -1,24 +1,186 @@
 import React, { FC } from "react";
 import { Typography } from "@mui/material";
 import { useFormatTableData } from "../../../hooks/";
+import { useParams } from "react-router-dom";
 import { Table } from "../../../components";
+import { Renderer } from "components/Renderer";
+import { useFormSubmit } from "hooks/useFormSubmit";
+import { useFormControls } from "hooks/useFormControls";
+import { GDXModal } from "components/GDXModal";
+import { apiAxios } from "utils";
+import { useQuery } from "react-query";
+import { ReadForm } from "components/ReadForm";
+import { EditForm } from "components/EditForm";
+import { IEditFields } from "types";
+import { Button } from "@mui/material";
+import { CreateForm } from "components/CreateForm";
 
 export const Subcontractors: FC = () => {
+  const {
+    handleEditMode,
+    handleOpen,
+    handleClose,
+    handleCurrentRowData,
+    handleFormType,
+    formType,
+    open,
+    editMode,
+    currentRowData,
+  } = useFormControls();
+
+  const { subcontractorId } = useParams();
+
   const { data, isLoading } = useFormatTableData({
     tableName: "subcontractors",
     apiEndPoint: "subcontractors",
+    handleClick: handleOpen,
   });
 
+  const { handlePost, handleUpdate, Notification } = useFormSubmit();
+
+  const getSubcontractors = async () => {
+    const subcontractors = await apiAxios().get(`/subcontractors/${currentRowData?.id}`);
+    return subcontractors.data.data[0];
+  };
+
+  // Queries
+  // todo: Define a good type. "Any" type temporarily permitted.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const subcontractorsQuery: any = useQuery(
+    `subcontractors - ${currentRowData?.id}`,
+    getSubcontractors,
+    {
+      refetchOnWindowFocus: false,
+      retryOnMount: false,
+      refetchOnReconnect: false,
+      retry: false,
+      staleTime: Infinity,
+    }
+  );
+
+  const readFields = [
```

Comment:
this won't work for you, you will have to change this for subcontractors

Reply:
You're right - I was going to change this when I linked the backend, but on reflection I'll change it now so as to avoid confusion for others.